### DISCUSSION TITLE
High Pressure properties for R-407A, R-410, r-507a [Bug]

### DISCUSSION THREAD

**Discussion by gcbrindle** [DanWBR/dwsim#443] [General]
Dan,

I'm trying to model these fluids up to 30MPa and 150C for industrial use.  Coolprop models is valid "Maximum temperature [K] | 500.0,
Maximum pressure [Pa] | 50000000.0, yet anytime i try to model heat exchanger or pipe with heat loss at that pressure, it won't converge.

I look at some cP vs temperature curves for the component in the interface and there curves don't make sense to me.

Can you check the coding please?

</body>
</html>

IMPORTANT: This section is for code-related bugs. For usage issues, please go to the [Support Forums](https://sourceforge.net/p/dwsim/discussion/).

**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Windows, Linux, macOS]
 - Version [e.g. v6.4.1]

**Additional context**

**Comment by DanWBR** [comment]
@gcbrindle can you attach a sample flowsheet?

**Comment by DanWBR** [comment]
@gcbrindle If you look at the temperature-dependent data for the refrigerants from CoolProp database, you'll notice in the comments section that they were fitted from CoolProp. The data shown here will only be used for non-CoolProp property packages, otherwise DWSIM will always use the values returned by CoolProp directly.

You'll notice for R404A that the fit for liquid density and thermal conductivity was not good, though this doesn't matter if you use CoolProp for the calculations:

![image](https://user-images.githubusercontent.com/2669419/218139291-5f5513b0-80f8-4eec-bc79-0f6356451b34.png)

**Comment by gcbrindle** [comment]
[](url)
[Test DWSIM for Daniel.zip](https://github.com/DanWBR/dwsim/files/10722959/Test.DWSIM.for.Daniel.zip)

**Comment by DanWBR** [comment]
@gcbrindle when DWSIM asks for R507A's saturation temperature at 4369503 Pa, it gives us the following error:

'The molar density of -58479902235639.117188 mol/m3 is below the minimum of 0.000000 mol/m3'

The maximum pressure reported for R507A by CoolProp is 50000000 Pa.

**Comment by gcbrindle** [comment]
So is there a work around?

**Comment by DanWBR** [comment]
@gcbrindle I'm not sure, but this seems to be a convergence issue with CoolProp, not necessarily a bug

**Comment by gcbrindle** [comment]
Maybe this belongs in a different forum but here's the question:  How do i use DWSim to model CO2, or R507 or anything else far above the critical point? my actual process is much more than the simplified one I sent.  Is it that i should use a different property package or do the reference files created by coolprop create a fault using any property package?  My target is heating and cooling  with friction losses in a process around 20 MPa and 100-140°C with low side temperatures of 30°C and 1.7 MPa

**Comment by DanWBR** [comment]
@gcbrindle one alternative I see is to use DWSIM Pro + REFPROP, you can get a trial and see if it works for your process. We could file a bug for CoolProp but they already have a lot of bugs to work on and are not very fast at fixing them, it might take a year or even more.
